packages/@aws-cdk/aws-lambda/lib/singleton-lambda.ts

```diff
@@ -61,6 +61,10 @@ export class SingletonFunction extends FunctionBase {
     this.grantPrincipal = this.lambdaFunction.grantPrincipal;
 
     this.canCreatePermissions = true; // Doesn't matter, addPermission is overriden anyway
+
+    if (this.lambdaFunction.isBoundToVpc) {
+      this._connections = this.lambdaFunction.connections;
+    }
```

Comment:
I believe this would be a better implementation.

```ts
class SingletonFunction .... {
  public get connections(): ec2.Connections {
    return this.lambdaFunction.connections;
  }
}
```

Reply:
Had to override `isBoundToVpc` as well since the base implementation checks `this._connections` which is not set for the SingletonFunction